src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10142,6 +10139,7 @@ private static bool CanFollowCast(SyntaxKind kind)
                 case SyntaxKind.MinusGreaterThanToken:
                 case SyntaxKind.QuestionQuestionToken:
                 case SyntaxKind.EndOfFileToken:
+                case SyntaxKind.EqualsGreaterThanToken:
```

Comment:
>case SyntaxKind.EqualsGreaterThanToken: [](start = 16, length = 39)

It is not obvious why this change is needed. Please add a comment and an example.

Reply:
Recorded in #23222.